### PR TITLE
R4R: Dynamic extra incentive to bsc relayer

### PR DESCRIPTION
**Pull request by HaoyangLiu**
Implement https://github.com/binance-chain/BEPs/pull/86

### CODE REVIEW

File: contracts/RelayerIncentivize.sol [hunk 1]
```diff
@@ -39,6 +39,8 @@ contract RelayerIncentivize is IRelayerIncentivize, System, IParamSubscriber {
 
   mapping(address => uint256) public relayerRewardVault;
 
+  uint256 public compensateRelayerAmount;
```
Comment (guagualvcha) [inline contracts/RelayerIncentivize.sol:42]: can we rename `compensate`?

File: contracts/RelayerIncentivize.sol [hunk 2]
```diff
@@ -64,6 +66,10 @@ contract RelayerIncentivize is IRelayerIncentivize, System, IParamSubscriber {
       actualAmount = ISystemReward(TOKEN_HUB_ADDR).claimRewards(address(uint160(INCENTIVIZE_ADDR)), amount);
     }
 
+    if (compensateRelayerAmount > 0) {
+        actualAmount = actualAmount.add(ISystemReward(SYSTEM_REWARD_ADDR).claimRewards(address(uint160(INCENTIVIZE_ADDR)), compensateRelayerAmount));
```
Comment (guagualvcha) [inline contracts/RelayerIncentivize.sol:70]: if `fromSystemReward `, can we do not do `claimRewards `  twice?

File: contracts/RelayerIncentivize.sol [hunk 3]
```diff
@@ -216,6 +222,11 @@ contract RelayerIncentivize is IRelayerIncentivize, System, IParamSubscriber {
       uint256 newCallerCompensationDenominator = BytesToTypes.bytesToUint256(32, value);
       require(newCallerCompensationDenominator != 0 && newCallerCompensationDenominator >= callerCompensationMolecule, "the newCallerCompensationDenominator must not be zero and no less than callerCompensationMolecule");
       callerCompensationDenominator = newCallerCompensationDenominator;
+    } else if (Memory.compareStrings(key,"compensateRelayerAmount")) {
```
Comment (guagualvcha) [inline contracts/RelayerIncentivize.sol:225]: rename  `compensateRelayerAmount` please

**Comment by guagualvcha** [comment]
Please add more description of this PR.

**Review by guagualvcha — DISMISSED** [review]

**Review by yutianwu — APPROVED** [review]

**Review by guagualvcha — APPROVED** [review]

**Review by mikegeorge3901 — APPROVED** [review]

**Pull request merged**